### PR TITLE
feat(search): #191 add grouping to search (common)

### LINKED ISSUE #191: Add groupings to search possibilities (common)

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Ajout de la base pour permettre la recherche de grouping dans les bars de recherche.

## Checklist tests
Rien

## Issue ticket number and link
[MA-894](https://jira.support-ent.fr/browse/MA-894)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts [hunk 1]
```diff
@@ -9,20 +10,31 @@ import {Group, GroupService} from "@common/services";
...
-    constructor(structureId: string, searchService: SearchService, groupService: GroupService) {
+    constructor(structureId: string, searchService: SearchService, groupService: GroupService, groupingService?: GroupingService) {
         super(structureId, searchService);
         this.groupService = groupService;
+        this.groupingService = groupingService;
     }
 
-    public getGroups() {
-        return this.groups;
+    //Return only group. Return group from grouping but not grouping
```
Comment (FlorentMr) [inline common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts]: J'aurais plus mis quelque chose du genre "Only returns group from grouping, but not the grouping itself" pour un peu plus de clarté

File: common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts [hunk 2]
```diff
@@ -45,22 +57,43 @@ export class GroupsSearch extends AutoCompleteUtils {
         this.selectedGroups = [];
     }
 
-    public selectGroups(valueInput, groupItem) {
+    public selectGroups(valueInput, groupItem: Group | Grouping) {
```
Comment (FlorentMr) [inline common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts]: Typage de valueInput

File: common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts [hunk 3]
```diff
@@ -45,22 +57,43 @@ export class GroupsSearch extends AutoCompleteUtils {
...
+            });
+        } else {
+            if (this.selectedGroups.find(group => group["id"] === groupItem.id) === undefined) {
+                this.selectedGroups.push(groupItem);
+            }
         }
     };
 
-    public selectGroup(valueInput, groupItem) {
+    public selectGroup(valueInput, groupItem: Group | Grouping) {
```
Comment (FlorentMr) [inline common/src/main/resources/ts/utils/autocomplete/groupsSearch.ts]: Typage de valueInput

File: common/src/main/resources/ts/utils/autocomplete/usersSearch.ts [hunk 4]
```diff
@@ -50,12 +50,21 @@ export class UsersSearch extends AutoCompleteUtils {
...
+    public async searchUsers(valueInput: string, profiles?: Array<string>) {
         try {
-            await Promise.all([
-                this.searchService.searchUser(this.structureId, valueInput, 'Personnel'),
-                this.searchService.searchUser(this.structureId, valueInput, 'Teacher')
-            ]).then(response => {
+            let promiseList: Array<Promise<User[]>> = []
+            if (!profiles) {
+                promiseList = [
+                    this.searchService.searchUser(this.structureId, valueInput, 'Personnel'),
```
Comment (FlorentMr) [inline common/src/main/resources/ts/utils/autocomplete/usersSearch.ts]: Magic string Personnel et Teacher

**Review by sjollois — APPROVED** [review]

**Review by FlorentMr — APPROVED** [review]

File: common/src/main/resources/ts/services/GroupingService.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+export interface GroupingService {
+    search(structureId: string, value: string): Promise<Grouping[]>;
+}
+
+export const GroupingService: GroupingService = {
+    search: async (structureId: string, value: string) => {
+        try {
+            value = value.replace("\\s", "").toLowerCase();
+            const {data} = await http.get(`/grouping/${structureId}/:id/list&searchValue=${value}`);
```
Comment (BastienBenard) [inline common/src/main/resources/ts/services/GroupingService.ts]: presences/grouping/${structureId}/:id/list&searchValue=${value}

File: common/src/main/resources/ts/model/Grouping.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import {Group} from "@common/services";
```
Comment (BastienBenard) [inline common/src/main/resources/ts/model/Grouping.ts]: Nommage du fichier

File: common/src/main/resources/ts/services/GroupingService.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import {Grouping, GroupingResponse} from "@common/model/Grouping";
```
Comment (BastienBenard) [inline common/src/main/resources/ts/services/GroupingService.ts]: Nommage fichier

**Pull request merged**